xds/internal/balancer/clusterresolver/priority_test.go

```diff
@@ -117,8 +117,10 @@ func (s) TestEDSPriority_HighPriorityReady(t *testing.T) {
 	edsb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Ready})
 
 	// Test roundrobin with only p0 subconns.
-	if err := testRoundRobinPickerFromCh(cc.NewPickerCh, []balancer.SubConn{sc1}); err != nil {
-		t.Fatal(err)
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	if err := cc.WaitForRoundRobinPicker(ctx, sc1); err != nil {
+		t.Fatal(err.Error())
```

Comment:
Replace with `t.Fatal(err)`.

Reply:
other usages of `WaitForRoundRobinPicker` seem to be using `err.Error()`. Making it to hard fail seems to be giving a panic. lmk what do you think?

Reply:
Spoke offline. both `Fatal(err)` and `Fatal(err.Error())` gives the same output. Decided to go with the shorter form [(?)](https://g3doc.corp.google.com/go/g3doc/style/decisions.md?cl=head#keep-going)